### PR TITLE
Feature: flatten struct

### PR DESCRIPTION
**Pull request by TomBurdge**
Add flatten struct function. Pure python (there's not much transforming in the data structure. structs are just series within series it's just moving the refernece to the series in the Frame namespace I believe).

Moved the snake case test to `test_transformations.py`

### CODE REVIEW

**Review by LyndonFan — CHANGES_REQUESTED** [review]
Code looks good, but how do we want to handle:

- columns which aren't structs? I guess this would just raise an error saying the `.struct` bit doesn't make sense, or we could check for it earlier and reject the process.
- structs with nested fields? Do we add a "recursive" option, or force the user to manually run this multiple times if they want it?
- structs whose values is a list? e.g. `{"name": "Charles", "nickname": ["Charlie", "Char"]}`. Breaking each value in the list into its own column doesn't sound clean (e.g. what if there's 1 element for one of the lists, then 1000 elements for another?), and the current method will just leave them as is. Actually a bit inclined to keep it as is, but wanted to bring this up.

**Comment by TomBurdge** [comment]
* Yeah, I can implement something simple to check it and give a better error message.
* Does polars have nested structs? I thought you said it didn't when we originally suggested, but happy to implement if it does with a recursive flag that is False by default. I will try to find out if `polars` does/doesn't have it.
* Yeah if we really wanted to do something like that maybe we could do it with a `_{poistion}` but in a separate namespace? But to be honest, I expect the use case where this would be okay practice would be highly specific.

**Comment by TomBurdge** [comment]
>     * columns which aren't structs? I guess this would just raise an error saying the `.struct` bit doesn't make sense, or we could check for it earlier and reject the process.

I'm starting to think that this should probably just error as is, because checking the dtype will just slow it down - checking the dtype  for a column in a LazyFrame requires processing a logical plan, which _can_ take a long time on complex queries.

**Comment by TomBurdge** [comment]
So nested structs are possible in polars, I'll implement with a recursive option:
```python
df = pl.DataFrame({ "coords": [{"x": {"z":1,"a":2}, "y": 4}, {"x": {"z":1,"a":2}, "y": 9}, {"x": {"z":1,"a":2}, "y": 16}], "multiply": [10, 2, 3],} )
print(df)

shape: (3, 2)
┌────────────┬──────────┐
│ coords     ┆ multiply │
│ ---        ┆ ---      │
│ struct[2]  ┆ i64      │
╞════════════╪══════════╡
│ {{1,2},4}  ┆ 10       │
│ {{1,2},9}  ┆ 2        │
│ {{1,2},16} ┆ 3        │
└────────────┴──────────┘
```

**Comment by TomBurdge** [comment]
Probably not the cleanest implementation, but I have added an argument which allows for recursive struct unnesting.

File: harley/transformations.py [hunk 1]
```diff
@@ -1,6 +1,55 @@
...
+            if isinstance(dtype, Struct)
+        ]):
+            df = flatten_struct(
+                df=df,
+                struct_columns=new_unnested_columns,
+                separator=separator,
+                drop_original_struct=drop_original_struct,
+                recursive=True,
+                limit=(limit - 1 if limit else None),
+            )
```
Comment (LyndonFan) [inline harley/transformations.py:54]: Works, but the recursive calls could be expensive if the struct is large. I might give it a go as well

**Review by LyndonFan — CHANGES_REQUESTED** [review]
Missing changes in `__init__.py`, and see comment on recursion, but rest lgtm

File: harley/transformations.py (hunk 1 above)
Comment (TomBurdge) [reply harley/transformations.py:54]: Yeah feel free! I think the limit helps a bit

**Pull request merged**